### ISSUE TITLE
[homework: doing, reading, watching] Linear, logistic regressions and embedding visualizations of Zulip data

### ISSUE THREAD

**Issue by indrekromet** [onefact/datathinking.org#115]
## Doing
- [x] Clean Data Thinking Zulip chat data, located at https://github.com/onefact/datathinking.org-codespace/blob/main/data/datathinking.zulipchat.com/raw/messages-000001.json - put it in a `polars` dataframe and compute summary statistics of the dataset
- [x] Analyze this Zulip chat data using logistic regression, linear regression, and embeddings with the tools we have learned in the lectures (don't forget to ask ChatGPT, Claude, Lex, GPT-4 for help as much as you need, and ask for help on the Data Thinking Zulip chat :)
- [x] Create a visualization of logistic regression of the Data Thinking Zulip chat data
- [x] Create a visualization of linear regression applied to the Data Thinking Zulip chat data 
- [x] Create a visualization of embeddings using the Data Thinking Zulip Chat data
- [x] Make a copy of the Overleaf template: https://www.overleaf.com/read/ghpyzqwqwxpv (need to create an account and/or sign in if this is your first time using Overleaf). To make a copy, open the project after signing in using this link, and click on `Menu`, then `Copy Project`:
<img width="310" alt="image" src="https://user-images.githubusercontent.com/5317244/226241154-a96f364e-6892-4229-b363-3a507b5dc2f2.png">

- [x] In Overleaf, edit the template and figure out how to include a PDF figure in the report, alongside a brief description (a few sentences or paragraphs is fine!) of each of the analyses you performed, why you chose them, and the math equation for the linear regression, logistic regression, and embedding you used.
- [x] Add the PDF of the report to this issue as a comment.
- [x] Send a message on Zulip with a link to this comment, alongside the image representing your favorite visualization

## Reviewing
- [x] Review how Jaan got unstuck in the lecture recordings at https://panopto.ut.ee/Panopto/Pages/Sessions/List.aspx?folderID=43bb180c-79a6-4324-b055-afa400ecd1a0
- [x] Review collaborative whiteboards from past classes: listed at https://www.datathinking.org/university-of-tartu
- [x] Review Jupyter notebooks from past classes: 
    - https://github.com/onefact/datathinking.org-codespace/blob/main/notebooks/230313-embeddings-volume-2.ipynb
    - https://github.com/onefact/datathinking.org-codespace/blob/main/notebooks/230316-embeddings-text-data-like-chat-logs-and-networks.ipynb
    - https://github.com/onefact/datathinking.org-codespace/tree/main/notebooks

## Reading
- [x] [inspiration for how to categorize discussion data in `json` format with chatgpt] https://genmon.github.io/braggoscope/about & https://news.ycombinator.com/item?id=35073603
- [x] [help with helping you learn to be a prompt engineer] https://github.com/dair-ai/Prompt-Engineering-Guide
- [x] [help with storytelling] https://themarkup.org/hello-world/2023/02/04/journalistic-lessons-for-the-algorithmic-age
- [x] [help with debugging] https://wizardzines.com/images/debugging/toc-letter.pdf
- [x] [design, visual communication] https://anthonyhobday.com/sideprojects/saferules/ & https://news.ycombinator.com/item?id=34684761
- [x] [background on ChatGPT and Claude from an ex-OpenAI, Anthropic founder] read `The Boy Whose Light Went Out` by Jack Clark http://techpolicylab.uw.edu/wp-content/uploads/2022/04/Telling_Stories_Pages_4-4-22.pdf 
- [x] [history of OpenAI internal difficulties] https://www.technologyreview.com/2020/02/17/844721/ai-openai-moonshot-elon-musk-sam-altman-greg-brockman-messy-secretive-reality/
- [x] [context for supply chain ban on GPU exports/imports] https://www.palladiummag.com/2021/10/11/the-triumph-and-terror-of-wang-huning/
- [x] [starting to think about bias in what you read from chatgpt or elsewhere] https://catalogofbias.org/biases/
- [x] [psychological tools for support in dealing with ChatGPT, especially if you decide to try jailbreaks] https://www.bellingcat.com/resources/2022/11/23/how-to-maintain-mental-hygiene-as-an-open-source-researcher/
- [x] [how data is being used by open source intelligence researchers in war] https://www.cbsnews.com/news/bellingcat-russia-putin-ukraine-60-minutes-2022-08-21/
- [x] [background on social media and recommendation algorithms] https://knightcolumbia.org/content/understanding-social-media-recommendation-algorithms

## Watching
(message Jaan if you need a VPN or these links don't work)
- [x] [context for popular media portrayals of chatgpt] South Park S26E04 - "Deep Learning" - https://southpark.cc.com/episodes/8byci4/south-park-deep-learning-season-26-ep-4 
- [x] [context for how the CEO of OpenAI talks about it] https://abcnews.go.com/Technology/video/openai-ceo-cto-risks-ai-reshape-society-97949497 [article](https://abcnews.go.com/Technology/openai-ceo-sam-altman-ai-reshape-society-acknowledges/story?id=97897122)
- [x] [an emotional history of the world; context for our data thinking practice of feel, ask, do, think -- and how the thoughts in people's heads can shape society] 1 episode of https://thoughtmaybe.com/cant-get-you-out-of-my-head/

**Comment by indrekromet** [comment]
[datathinking_org_latex_report_template__Copy_-indrek.pdf](https://github.com/onefact/datathinking.org/files/11132786/datathinking_org_latex_report_template__Copy_-indrek.pdf)

**Comment by ikr503** [comment]
critique from me: The visualizations make sense, I believe, but I don't have the deepest knowledge of why or how much they truly make sense. I could've asked somewhere for feedback on these. I could've looked further into the Google TensorFlow before finishing the homework. I'd consider it an honest job of a homework, that showcases the genuine knowledge I have, as well the clearly apparent gaps. I definitely did some asking, thinking, feeling and doing in this homework. I personally appreciate the brevity of my work, no need to do more than is needed, in my opinion.

**Comment by ikr503** [comment]
and ChatGPT-4 coming in hard with the following:

Title and Abstract: These are okay but the abstract could use a bit more context. What's the goal of this analysis? What's the significance of using chat data for these models?

Introduction: This section is lacking in context and clarity. It is important to clarify what linear regression, logistic regression, and embeddings are used for in this context. Why did you choose these particular methods for this analysis? What are you hoping to learn or prove?

Section 2 - Logistic Regression: Your explanation is mostly clear but you should try to explain what the mathematical equation represents in the context of your analysis. For instance, what does each term represent? And why is the AUC score not 1? A perfect model is highly unlikely in real-world situations and you should acknowledge this. Furthermore, it's crucial to present the results visually or at least provide some insights into the key features that contributed to the presence of a URL.

Section 3 - Linear Regression: Your R2 value of 1 should raise a flag. An R2 of 1 indicates a perfect fit, which is usually unrealistic in real-life scenarios. It's more likely that the model is overfitting the data. Explain how you ensured this was not the case or if you didn't, how you plan on improving this in future analyses. Like the previous section, a visualization or at least some insights into the features would be helpful.

Section 4 - Embeddings: It is quite unclear what this section is about. You should provide more explanation about what the Universal Sentence Encoder (USE) does and why you chose it. Why did you want to create embeddings? What are they typically used for? It is also crucial to interpret the embeddings. Saying it "looks cool" doesn't provide any useful insights. You should attempt to interpret your findings or, if you're unsure, at least propose further analysis.

Visualizations: Without being able to see your visualizations, it's hard to critique them. However, remember that all visualizations should have clear labels, titles, and a legend if necessary. They should be clear enough to understand without much additional explanation.

General feedback: It's good to see the use of machine learning techniques on chat data, however, the report lacks explanation and insight. In data science, being able to interpret and communicate your findings is just as important as creating the models. Also, don't forget to discuss any limitations or assumptions in your analysis.

Remember to write for an audience that might not be familiar with your topic. Define your terms, explain why you're doing what you're doing, and make sure your report tells a clear story.
